### DISCUSSION TITLE
Mayavi (3D scientific visualization) integration with nicegui

### DISCUSSION THREAD

**Discussion by anandjebakumar** [zauberzeug/nicegui#3150] [Ideas / Feature Requests]
How easy/difficult would it be to integrate mayavi with nicegui? I really like the ability to integrate Matplotlib/Plotly plots and looks like three.js integration is also there to help with 3D scenes. Integrating mayavi would be a huge benefit for complex 3D scientific visualizations. Thanks for all you do!

**Comment by falkoschindler** [comment]
Hi @anandjebakumar,

I'm not sure about a NiceGUI integration of [Mayavi](https://docs.enthought.com/mayavi/mayavi/). We would somehow need to extract the VTK rendering data and send it to the client. For static images this shouldn't be a problem. But for interactivity, we would need to run something like [vkt.js](https://kitware.github.io/vtk-js/index.html) on the client and communicate with the server.

**Reply by anandjebakumar** [reply]
How easy/difficult is this? Asking to see if I can contribute and if there's community interest in it.

**Reply by falkoschindler** [reply]
I don't think it's easy. But I don't know Mayavi and haven't worked with VTK yet, so I can't really tell how hard it might be.
